### ISSUE TITLE
LLaDA inference logits differ across NVIDIA GPUs (Blackwell vs 4090) with same weights/code

### ISSUE THREAD

**Issue by Kevin-lkw** [ML-GSAI/LLaDA#131]
Hi, thanks for the great work on LLaDA. I found an inference consistency issue across GPUs for the same model/codepath.

Setup:

Model: GSAI-ML/LLaDA-8B-Instruct

PEFT loading:
```code
model = PeftModel.from_pretrained(
    model,
    checkpoint_path,
    torch_dtype=torch.bfloat16,
).to(local_rank).eval()
```
GPUs tested:
RTX 4090: Hopper
Pro 6000: Blackwell 

Observation:

With the same weights, inputs, and code, logits = model(x).logits already differs across the two GPUs.
The difference is typically around 0.05. 
Example output dtype is bfloat16 when autocast is disabled.
This mismatch later gets amplified by softmax/topk in generation, causing decoding divergence. we eval math500 dataset and get a +2%  performance gain in Pro6000.

What I already tried:

torch.autocast(device_type="cuda", enabled=False)
sdpa_kernel([SDPBackend.MATH])
disabled TF32
disabled bf16 reduced-precision reduction
deterministic algorithms enabled

Even after these changes, the logits are still not aligned across the two GPUs.

Question:

Is cross-GPU inconsistency at this level expected for LLaDA inference in bf16?
Are there any recommended settings to improve numerical alignment across architectures?
Has this been observed before on Hopper/Blackwell-class GPUs?

Thanks.
